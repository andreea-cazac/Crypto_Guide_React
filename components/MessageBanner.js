import React from 'react';
import {StyleSheet, Text, View} from 'react-native';
import {GlobalStyle} from '../constants/GlobalStyle';

const MessageBanner = ({ type = 'info', title, message }) => {
    const isError = type === 'error';
    const isSuccess = type === 'success';

    return (
        <View
            style={[
                styles.container,
                isError && styles.error,
                isSuccess && styles.success,
            ]}
        >
            {title ? <Text style={styles.title}>{title}</Text> : null}
            <Text style={styles.message}>{message}</Text>
        </View>
    );
};

const styles = StyleSheet.create({
    container: {
        padding: 14,
        borderRadius: 10,
        marginVertical: 10,
        backgroundColor: GlobalStyle.colors.background,
        borderWidth: 1,
        borderColor: GlobalStyle.colors.primary,
    },
    error: {
        backgroundColor: GlobalStyle.colors.background,
        borderColor: GlobalStyle.colors.errorColor,
    },
    success: {
        backgroundColor: GlobalStyle.colors.background,
        borderColor: GlobalStyle.colors.successColor,
    },
    title: {
        fontSize: 18, // bigger
        fontWeight: 'bold',
        color: GlobalStyle.colors.primary,
        marginBottom: 4,
    },
    message: {
        fontSize: 14,
        fontWeight: '400',
        color: GlobalStyle.colors.primary,
    },
});

export default MessageBanner;