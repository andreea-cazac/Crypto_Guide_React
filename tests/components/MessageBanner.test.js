import React from 'react';
import { render } from '@testing-library/react-native';
import MessageBanner from '../../components/MessageBanner';

describe('MessageBanner Component', () => {
    const title = 'Test Title';
    const message = 'This is a test message.';

    it('renders correctly with title and message', () => {
        const { getByText, toJSON } = render(
            <MessageBanner title={title} message={message} />
        );
        // Verify both title and message are rendered.
        expect(getByText(title)).toBeTruthy();
        expect(getByText(message)).toBeTruthy();
        // Snapshot test for regression.
        expect(toJSON()).toMatchSnapshot();
    });

    it('renders correctly without a title', () => {
        const { queryByText, getByText, toJSON } = render(
            <MessageBanner message={message} />
        );
        // Message should be there, title should not.
        expect(getByText(message)).toBeTruthy();
        expect(queryByText(title)).toBeNull();
        expect(toJSON()).toMatchSnapshot();
    });

    it('applies error styling when type is "error"', () => {
        const { toJSON } = render(
            <MessageBanner type="error" title={title} message={message} />
        );
        // The snapshot should reflect error styling (e.g., red background).
        expect(toJSON()).toMatchSnapshot();
    });

    it('applies success styling when type is "success"', () => {
        const { toJSON } = render(
            <MessageBanner type="success" title={title} message={message} />
        );
        // The snapshot should reflect success styling (e.g., green background).
        expect(toJSON()).toMatchSnapshot();
    });
});