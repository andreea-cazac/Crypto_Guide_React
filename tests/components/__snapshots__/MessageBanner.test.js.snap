// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MessageBanner Component applies error styling when type is "error" 1`] = `
<View
  style={
    [
      {
        "backgroundColor": "#fff",
        "borderColor": "#14213D",
        "borderRadius": 10,
        "borderWidth": 1,
        "marginVertical": 10,
        "padding": 14,
      },
      {
        "backgroundColor": "#fff",
        "borderColor": "#cc0000",
      },
      false,
    ]
  }
>
  <Text
    style={
      {
        "color": "#14213D",
        "fontSize": 18,
        "fontWeight": "bold",
        "marginBottom": 4,
      }
    }
  >
    Test Title
  </Text>
  <Text
    style={
      {
        "color": "#14213D",
        "fontSize": 14,
        "fontWeight": "400",
      }
    }
  >
    This is a test message.
  </Text>
</View>
`;

exports[`MessageBanner Component applies success styling when type is "success" 1`] = `
<View
  style={
    [
      {
        "backgroundColor": "#fff",
        "borderColor": "#14213D",
        "borderRadius": 10,
        "borderWidth": 1,
        "marginVertical": 10,
        "padding": 14,
      },
      false,
      {
        "backgroundColor": "#fff",
        "borderColor": "#00aa00",
      },
    ]
  }
>
  <Text
    style={
      {
        "color": "#14213D",
        "fontSize": 18,
        "fontWeight": "bold",
        "marginBottom": 4,
      }
    }
  >
    Test Title
  </Text>
  <Text
    style={
      {
        "color": "#14213D",
        "fontSize": 14,
        "fontWeight": "400",
      }
    }
  >
    This is a test message.
  </Text>
</View>
`;

exports[`MessageBanner Component renders correctly with title and message 1`] = `
<View
  style={
    [
      {
        "backgroundColor": "#fff",
        "borderColor": "#14213D",
        "borderRadius": 10,
        "borderWidth": 1,
        "marginVertical": 10,
        "padding": 14,
      },
      false,
      false,
    ]
  }
>
  <Text
    style={
      {
        "color": "#14213D",
        "fontSize": 18,
        "fontWeight": "bold",
        "marginBottom": 4,
      }
    }
  >
    Test Title
  </Text>
  <Text
    style={
      {
        "color": "#14213D",
        "fontSize": 14,
        "fontWeight": "400",
      }
    }
  >
    This is a test message.
  </Text>
</View>
`;

exports[`MessageBanner Component renders correctly without a title 1`] = `
<View
  style={
    [
      {
        "backgroundColor": "#fff",
        "borderColor": "#14213D",
        "borderRadius": 10,
        "borderWidth": 1,
        "marginVertical": 10,
        "padding": 14,
      },
      false,
      false,
    ]
  }
>
  <Text
    style={
      {
        "color": "#14213D",
        "fontSize": 14,
        "fontWeight": "400",
      }
    }
  >
    This is a test message.
  </Text>
</View>
`;
